paprika/restraints/plumed.py

```diff
@@ -0,0 +1,348 @@
+        """
+        return self._units
+
+    @units.setter
+    def units(self, value):
+        self._units = value
+
+    def __init__(self):
+        self._file_name = "plumed.dat"
+        self._restraint_list = None
+        self._window_list = None
+        self._path = "./"
+        self._uses_legacy_k = True
+        self.k_factor = 1.0
+        self._units = None
+        self.header_line = None
+        self.group_index = None
+        self.group_atoms = None
+
+    def _initialize(self):
+        # Set factor for spring constant
+        if self.uses_legacy_k:
+            self.k_factor = 2.0
+
+        # Check units
+        # NOTE: We have to resort to strings until (if) we migrate all quantities to Pint/SimTK units
+        if self.units is None:
+            self.units = {"energy": "kcal/mol", "length": "A", "time": "ns"}
+        _check_plumed_units(self.units)
+
+        # header line
+        self.header_line = f"UNITS LENGTH={self.units['length']} ENERGY={self.units['energy']} TIME={self.units['time']}"
+
+    def dump_to_file(self):
+
+        self._initialize()
+
+        # Loop over APR windows
+        for windows in self.window_list:
+
+            window, phase = parse_window(windows)
+
+            # Check if file exist and write header line
+            with open(os.path.join(self.path, windows, self.file_name), "w") as file:
+                file.write(self.header_line + "\n")
+
+            cv_index = 1
+            cv_list = []
+            bias_list = []
+
+            self.group_index = 1
+            self.group_atoms = {}
+
+            # Parse each restraint in the list
+            for restraint in self.restraint_list:
+                # Skip restraint if the target or force constant is not defined.
+                # Example: wall restraints only used during the attach phase.
+                try:
+                    target = restraint.phase[phase]["targets"][window]
+                    force_constant = (
+                        restraint.phase[phase]["force_constants"][window]
+                        * self.k_factor
+                    )
+                except TypeError:
+                    continue
+
+                # Convert list to comma-separated string
+                atom_index = self._get_atom_indices(restraint)
+                atom_string = ",".join(map(str, atom_index))
+
+                # Determine collective variable type
+                colvar_type = "distance"
+                if len(atom_index) == 3:
+                    colvar_type = "angle"
+                    target *= _PI_ / 180.0
+                elif len(atom_index) == 4:
+                    colvar_type = "torsion"
+                    target *= _PI_ / 180.0
+
+                # Determine bias type for this restraint
+                bias_type = get_bias_potential_type(restraint, phase, window)
+                if bias_type is "harmonic":
```

Comment:
Are you sure you want `is` here? I think this can be dangerous because you're actually testing if they point to the same thing rather than whether the strings are the same.

Reply:
Also, I think it's a little unclear that `bias_type` is first `"harmonic"` but then `"restraint"`.

Reply:
>Are you sure you want is here? 

Good point, `==` would be the better choice when comparing strings.

>Also, I think it's a little unclear that bias_type is first `"harmonic"` but then `"restraint"`.

I was trying to make it general and label it as `harmonic` (the keyword in Plumed is `restraint`). I was thinking ahead of myself, I had in mind of also writing a plugin for the `Colvar` module used in `NAMD` and `LAMMPS`. I can just make it `restraint` by default and worry about it later when I write the `Colvar` plugin.

Reply:
👍